winpr/libwinpr/utils/sam.c

```diff
@@ -89,6 +89,9 @@ static BOOL SamLookupStart(WINPR_SAM* sam)
 	size_t readSize;
 	INT64 fileSize;
 
+	if (!sam || !sam->fp)
+		return FALSE;
```

Comment:
Checking sam is not needed (see callstacks of  SamLookupStart).
About ```sam->fp```, I guess the logic should be changed in ```SamOpen``` to not return a WINPR_SAM when the file can't be opened (and os that check would not be needed).